Tests/PointFreeTests/NotFoundMiddlewareTests.swift

```diff
@@ -0,0 +1,44 @@
+import Either
+import Html
+import HtmlPrettyPrint
+import HttpPipeline
+@testable import PointFree
+import PointFreeTestSupport
+import Prelude
+import Optics
+import SnapshotTesting
+import XCTest
+#if !os(Linux)
+  import WebKit
+#endif
+
+final class NotFoundMiddlewareTests: TestCase {
+  override func setUp() {
+    super.setUp()
+    AppEnvironment.push(\.database .~ .mock)
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    AppEnvironment.pop()
+  }
+
+  func testNotFound() {
+    let result = connection(from: URLRequest(url: URL(string: "http://localhost:8080/404")!))
+      |> siteMiddleware
+      |> Prelude.perform
+
+    assertSnapshot(matching: result)
```

Comment:
Just wondering if we want to test logged-in/subscribed states or if that's overkill.

Reply:
yeah might as well!